soh/soh/Enhancements/randomizer/3drando/fill.cpp

```diff
@@ -488,14 +488,16 @@ static void GeneratePlaythrough() {
 RandomizerArea LookForExternalArea(Area* curRegion, std::vector<RandomizerRegion> alreadyChecked){//RANDOTODO curREGION
   for (auto& entrance : curRegion->entrances) {
     RandomizerArea otherArea = entrance->GetParentRegion()->GetArea();
-    if(otherArea != RA_NONE){
-      return otherArea;
-      //if the area hasn't already been checked, check it
-    } else if (std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()) {
-      alreadyChecked.push_back(entrance->GetParentRegionKey());
-      RandomizerArea passdown = LookForExternalArea(entrance->GetParentRegion(), alreadyChecked);
-      if(passdown != RA_NONE){
-        return passdown;
+    if (otherArea != RA_LINKS_POCKET){ //if it's links pocket, do not propogate this, link's pocket is not a real Area
+      if(otherArea != RA_NONE){
+        return otherArea;
+        //if the area hasn't already been checked, check it
+      } else if (std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()) {
+        alreadyChecked.push_back(entrance->GetParentRegionKey());
```

Comment:
Ok, so you want to modify a parameter, and pass it recursively. I would suggest making a copy just up here, not the whole parameter, passed to a function.
```
auto checkedAreasCopy = alreadyChecked;
checkedAreasCopy .push_back(...)
```

Reply:
With alreadyChecked being a reference, additions are now kept going backwards which is a good thing as the point of the vector is to avoid processing the same area twice in the same search.Copying it would ruin this.